modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientTable.java

```diff
@@ -485,22 +488,23 @@ private <T> CompletableFuture<T> doSchemaOutInOpAsync(
                 .thenCompose(v -> {
                     ClientSchema schema = schemaFut.getNow(null);
 
-                    PartitionMapping forCrd = getPreferredNodeName(tableId(), provider, partitionsFut.getNow(null), schema, true);
-
-                    return ClientLazyTransaction.ensureStarted(tx, ch, forCrd).thenCompose(tx0 -> {
-                        @Nullable PartitionMapping forOp = getPreferredNodeName(tableId(), provider, partitionsFut.getNow(null), schema,
-                                tx0 == null); // Force coordinator mode for implicit transactions.
+                    Supplier<PartitionMapping> sup =
+                            () -> getPreferredNodeName(tableId(), provider, partitionsFut.getNow(null), schema, true);
+                    return ensureStarted(tx, ch, sup).thenCompose(tx0 -> {
+                        // Force coordinator mode for implicit transactions.
+                        @Nullable PartitionMapping forOp =
+                                getPreferredNodeName(tableId(), provider, partitionsFut.getNow(null), schema, tx0 == null);
 
                         WriteContext ctx = new WriteContext();
-                        ctx.pm = forOp;
+                        // Force proxy mode for requests collocated with coordinator to reduce passed enlistment info on commit.
```

Comment:
```suggestion
                        // Force proxy mode for requests colocated with coordinator to reduce passed enlistment info on commit.
```

I think we agreed on this variant.